### PR TITLE
Add org.gnome.Keysign

### PR DESCRIPTION
**Pull request by RyuzakiKK**

### CODE REVIEW

File: org.gnome.Keysign.yml [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+- --socket=x11
+- --share=network
+- --socket=system-bus
+- --device=all
+- --filesystem=~/.gnupg:ro
+- --filesystem=xdg-run/gnupg:ro
+# We're waiting for webcam support: https://github.com/flatpak/xdg-desktop-portal/issues/38;
+# we use --device=all meanwhile. We do network, because we're opening a port.
+appstream-compose: false
+tags:
```
Comment (nedrichards) [inline org.gnome.Keysign.yml]: flatpak is only for stable apps, please remove the nightly tag and prefix.

**Comment by nedrichards** [comment]
bot, build org.gnome.Keysign

File: org.gnome.Keysign.yml (hunk 1 above)
Comment (RyuzakiKK) [reply org.gnome.Keysign.yml]: Ah that was a leftover. I removed it.

File: org.gnome.Keysign.yml [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+      PYTHONUSERBASE: "/app"
+  ensure-writable:
+  - easy-install.pth
+  - setuptools.pth
+  sources:
+  - type: archive
+    url: https://pypi.python.org/packages/32/3c/e853a68b703f347f5ed86585c2dd2828a83252e1216c1201fa6f81270578/setuptools-26.1.1.tar.gz#md5=0744ee90ad266fb117d59f94334185d0
+    sha256: 475ce28993d7cb75335942525b9fac79f7431a7f6e8a0079c0f2680641379481
+  - type: script
+    dest-filename: configure
```
Comment (barthalion) [inline org.gnome.Keysign.yml]: You should use `no-autogen` and separate `Makefile` instead.

File: org.gnome.Keysign.yml [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+    type: archive
+    url: https://gstreamer.freedesktop.org/src/gst-python/gst-python-1.9.1.tar.xz
+    sha256: 2ab16a84efbabacfcf8f7920ab6c6725c577f08a82bf4cee45edf415a917015f
+- name: pip2
+  disabled: true
+  no-autogen: true
+  build-options:
+    # damnit, I don't know why it requires network...
+    build-args:
+    - "--share=network"
```
Comment (barthalion) [inline org.gnome.Keysign.yml]: There is no network access on Flathub builders.

**Comment by barthalion** [comment]
We have [YAML style guide](https://github.com/flathub/flathub/wiki/YAML-Style-Guide) now, would be nice to follow it here.

**Comment by RyuzakiKK** [comment]
Thanks for the feedback. With my last commit I should have implemented all the required modifications.

File: org.gnome.Keysign.yml [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+        url: https://gstreamer.freedesktop.org/src/gst-plugins-base/gst-plugins-base-1.12.3.tar.xz
+        sha256: d3d37b8489d37fa0018973d850bd2067b98af335fef2fa543ee7d40359e3cea5
+
+  - name: gst-plugins-good
+    sources:
+      - type: archive
+        url: https://gstreamer.freedesktop.org/src/gst-plugins-good/gst-plugins-good-1.12.3.tar.xz
+        sha256: 13e7f479296891fef5a686438f20ba7d534680becf2269ecc5ee24aa83b45f03
+
+  - name: gst-plugins-ugly
```
Comment (TingPing) [inline org.gnome.Keysign.yml]: I feel like a keysign tool doesn't need every codec gstreamer offers. Limiting it to the few plugins actually needed would be nice.

File: pip-Makefile [hunk 5]
```diff
@@ -0,0 +1,7 @@
+all:
```
Comment (TingPing) [inline pip-Makefile]: Just use the `simple` buildsystem for everything and avoid the makfiles.

**Comment by nedrichards** [comment]
bot, build org.gnome.Keysign

File: org.gnome.Keysign.yml [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+  - name: gst-plugins-bad
+    build-options:
+      config-opts:
+        - "--enable-zbar"
+    sources:
+      - type: archive
+        url: https://gstreamer.freedesktop.org/src/gst-plugins-bad/gst-plugins-bad-1.12.3.tar.xz
+        sha256: 36d059761852bed0f1a7fcd3ef64a8aeecab95d2bca53cd6aa0f08054b1cbfec
+    modules:
+      - name: zbar
```
Comment (nedrichards) [inline org.gnome.Keysign.yml]: This needs to be reconfigured to compile on aarch64. Here's an example for 0.10 https://github.com/flathub/com.github.bilelmoussaoui.Authenticator/blob/efc4b2f25b3aa3d7f62496e9c2ece91fb0180d9e/com.github.bilelmoussaoui.Authenticator.json#L26 or I think it just works in 0.20 https://github.com/flathub/com.github.paolostivanin.OTPClient/blob/2ebb8605652c34bccd545df15bc1eb9ad9c708cc/com.github.paolostivanin.OTPClient.json#L57

**Comment by RyuzakiKK** [comment]
I removed the gst ugly because it was no more needed, I removed the separate makefile and updated to zbar 0.20.

File: org.gnome.Keysign.yml [hunk 7]
```diff
@@ -0,0 +1,306 @@
+id: org.gnome.Keysign
+branch: master
```
Comment (barthalion) [inline org.gnome.Keysign.yml]: It should be `stable` or removed completely.

File: org.gnome.Keysign.yml [hunk 8]
```diff
@@ -0,0 +1,306 @@
...
+runtime: org.gnome.Platform
+runtime-version: '3.28'
+sdk: org.gnome.Sdk
+command: gnome-keysign
+copy-icon: true
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --share=network
+  - --socket=system-bus
```
Comment (barthalion) [inline org.gnome.Keysign.yml]: Why does it need access to entire system bus?

Comment (RyuzakiKK) [reply org.gnome.Keysign.yml]: Now I listed only the single bus we need (Avahi).

File: org.gnome.Keysign.yml (hunk 7 above)
Comment (RyuzakiKK) [reply org.gnome.Keysign.yml]: Removed with my last commit.

**Comment by barthalion** [comment]
bot, build org.gnome.Keysign

**Comment by nedrichards** [comment]
The appdata isn't being put into the correct place.

also:

> WARNING: Binary not found for Exec line in /srv/buildbot/worker/build-x86_64/org.gnome.Keysign/builddir/export/share/applications/org.gnome.Keysign.desktop: python -m keysign
Commit: 8ee50ecb2b0b7fb3c55abb8af3504c6a972a491a395a9e107594282b0dc8f9a4

Looks like there's some work to do here around the metadata.

**Comment by nedrichards** [comment]
So, if the app launches fine when installed (from both the shell and the command line) it's harmless, don't worry. I haven't built the app locally myself to test.

File: org.gnome.Keysign.yml [hunk 9]
```diff
@@ -0,0 +1,306 @@
...
+      - setuptools.pth
+    build-commands:
+      - python2 setup.py build
+      - python2 setup.py install --prefix=${FLATPAK_DEST}
+    sources:
+      - type: archive
+        url: https://pypi.python.org/packages/32/3c/e853a68b703f347f5ed86585c2dd2828a83252e1216c1201fa6f81270578/setuptools-26.1.1.tar.gz#md5=0744ee90ad266fb117d59f94334185d0
+        sha256: 475ce28993d7cb75335942525b9fac79f7431a7f6e8a0079c0f2680641379481
+
+  - name: pip
```
Comment (TingPing) [inline org.gnome.Keysign.yml]: Do you need `pip` at runtime? If not its already in the Sdk.

Comment (RyuzakiKK) [reply org.gnome.Keysign.yml]: No I don't but the pip2 version inside the Sdk is the old 7.1.0 and it doesn't allow the usage of `--prefix`.
You can easily try it with this manifest https://ptpb.pw/Nuee

Installing a newer pip version solved that problem.

**Comment by RyuzakiKK** [comment]
I bumped the manifest with the latest gnome-keysign release. Now also the appstream should compile as expected.

**Comment by nedrichards** [comment]
bot, build org.gnome.Keysign

**Comment by nedrichards** [comment]
Wrong checksum https://flathub.org/builds/#/builders/26/builds/206

**Comment by nedrichards** [comment]
bot, build org.gnome.Keysign

**Comment by barthalion** [comment]
Repository has been created:  https://github.com/flathub/org.gnome.Keysign

Thanks!

**Comment by RyuzakiKK** [comment]
Thanks @barthalion , can you also add @muelli with the write permissions?

**Comment by barthalion** [comment]
Done.

**Pull request closed without merging**